src/org/traccar/FilterHandler.java

```diff
@@ -227,6 +227,14 @@ private boolean filter(Position position) {
         }
 
         if (filterType.length() > 0) {
+            if (last != null && !filterZero(last) && !filterInvalid(last)) {
+                if (position.getLong(Position.KEY_STATUS) != last.getLong(Position.KEY_STATUS)) {
```

Comment:
This seems to be duplication of existing functionality from `skipAttributes`.

Reply:
As far as I remember I tried with skipAttributes but this resulted in invalid / zero GPS coordinates to be processed.
In our case the tracker goes to sleep and when it wakes up due to movement it sends an alarm message immediately, even if it does not have a fix yet. The alarm event needs to be forwarded to our push handler to inform the user, even if the fix is invalid / zero coordinates.

Reply:
Events work just fine with invalid and zero coordinates.

Reply:
But will it prevent the zero coordinate from being added to the route of the tracker?

Reply:
Now I did some further testing using various configuration options and 3.17 without the modifications in FilterHandler as in this PR.
The goal is to always update the status and attributes for the alarm, but use the last known position instead of an invalid / zero position:

I verified it using this LKGPS protocol heartbeat message with zero coordinates:
```
HEX: 2a48512c343731303030383838352c56312c3136333833312c562c303030302e303030302c4e2c30303030302e303030302c452c3030302e30302c3030302c3039303631382c46464535464246462c3233322c30312c666634312c302c3423
ASCII: *HQ,4710008885,V1,163831,V,0000.0000,N,00000.0000,E,000.00,000,090618,FFE5FBFF,232,01,ff41,0,4#
```

#### coordinates.maxError

I thought the `coordinates.maxError` setting should do the trick. Reading the documentation I understood that whenever the distance is greater than the value and position is invalid, the old position is used.

The relevant settings in the config file:
```xml
    <entry key='coordinates.filter'>true</entry>
    <entry key='coordinates.minError'>5</entry>
    <entry key='coordinates.maxError'>5</entry>
```

Still the message gets fully processed and the location jumps to zero. Either my understanding is wrong, or the message is not read as invalid.

#### filter.skipAttributes

Second attempt was to use `filter.zero` with `filter.skipAttributes`:
```xml
    <entry key='filter.zero'>true</entry>
    <entry key='filter.skipAttributes.enable'>true</entry>
    <entry key='filter.skipAttributes'>alarm,armed,battery,charge,result,status</entry>
```

Again location jumps to zero.
Using only `filter.zero` without `filter.skipAttributes` filters the message, but then also the alarm is not processed.

What else could I try to get the same behaviour as my patch?

Reply:
Skip attributes are applied before filtering, so if everything configured correctly positions with listed attributes won't be skipped.

Reply:
If I use skip attributes, it won't be skipped, that is correct. But if the message comes with zero coordinates they will be used too, that is the problem. The attributes should always be updated, but the coordinates only if they are non-zero, otherwise the previous coordinates should be used.
How to configure that?
Thanks!